UnchainedLauncher.Core/Services/Processes/Chivalry/LaunchPreparers/UnchainedContentPreparer.cs

```diff
@@ -0,0 +1,149 @@
+﻿using log4net;
+using log4net.Core;
+using Semver;
+using UnchainedLauncher.Core.JsonModels.Metadata.V3;
+using UnchainedLauncher.Core.Services.Mods;
+using UnchainedLauncher.Core.Utilities;
+using static LanguageExt.Prelude;
+
+namespace UnchainedLauncher.Core.Services.Processes.Chivalry.LaunchPreparers {
+    public class UnchainedContentPreparer: IChivalry2LaunchPreparer {
+        private readonly ILog logger = LogManager.GetLogger(typeof(UnchainedContentPreparer));
+
+        public UnchainedContentPreparer(Func<bool> getShouldUpdateDependencies, IModManager modManager, IReleaseLocator pluginReleaseLocator, IVersionExtractor fileVersionExtractor, IUserDialogueSpawner userDialogueSpawner) {
+            GetShouldUpdateDependencies = getShouldUpdateDependencies;
+            ModManager = modManager;
+            PluginReleaseLocator = pluginReleaseLocator;
+            FileVersionExtractor = fileVersionExtractor;
+            UserDialogueSpawner = userDialogueSpawner;
+        }
+
+        private Func<bool> GetShouldUpdateDependencies { get; }
+        private IModManager ModManager { get; }
+        private IReleaseLocator PluginReleaseLocator { get; }
+        private IVersionExtractor FileVersionExtractor { get; }
+        private IUserDialogueSpawner UserDialogueSpawner { get; }
+        
+        public async Task<bool> PrepareLaunch() {
```

Comment:
This all came from the PrepareUnchainedLaunch from the unchained launcher implementation